pkg/adapter/containers_remote.go

```diff
@@ -464,19 +464,33 @@ func (r *LocalRuntime) Run(ctx context.Context, c *cliconfig.RunValues, exitCode
 	results := shared.NewIntermediateLayer(&c.PodmanCommand, true)
 	cid, err := iopodman.CreateContainer().Call(r.Conn, results.MakeVarlink())
 	if err != nil {
-		return 0, err
+		return exitCode, err
 	}
 	if c.Bool("detach") {
 		_, err := iopodman.StartContainer().Call(r.Conn, cid)
 		fmt.Println(cid)
-		return 0, err
+		return exitCode, err
 	}
 	errChan, err := r.attach(ctx, os.Stdin, os.Stdout, cid, true, c.String("detach-keys"))
 	if err != nil {
-		return 0, err
+		return exitCode, err
+	}
+	err = <-errChan
```

Comment:
How about here, could this hang, forever waiting if the container has already exited (before the attach)?